web/src/features/panels/LeftPanel.tsx

```diff
@@ -117,37 +61,12 @@ function OuterPanel({ children }: { children: React.ReactNode }) {
       } ${isOpen ? '' : '-translate-x-full'}`}
     >
       {isMobile && <MobileHeader />}
-      <section className="h-full w-full">{children}</section>
+      <section className="h-full w-full">
+        <Suspense fallback={<LoadingSpinner />}>
+          {isZonePath ? <ZoneDetails /> : <RankingPanel />}
+        </Suspense>
```

Comment:
While this works would it not be better to use a outlet and pass the component down from the router?

https://reactrouter.com/en/main/components/outlet#outlet